api/src/services/applications/submit_application.py

```diff
@@ -35,6 +37,45 @@ def submit_application(db_session: db.Session, application_id: UUID) -> Applicat
             ],
         )
 
+    # Check if competition is closed
```

Comment:
As we're going to continue adding more and more validation rules, could we put each of these different checks into a function that we call here?

Then this submit function can be something like:

```
    logger.info("Processing application submit")
    application = get_application(db_session, application_id)

    validate_application_in_progress(application)
    validate_competition_open(application)
    # ... many many more we'll add soon

    application.application_status = ApplicationStatus.SUBMITTED
    logger.info("Application successfully submitted")

```
Also could then make tests that call those specific functions if we don't want to fully setup an API call test for each scenario we add/test.